### ISSUE TITLE
Cannot add or update a child row on comments

### ISSUE THREAD

**Issue by ddu17** [centreon/centreon-broker#32]
Hello,

I'm working with CBD 2.9.2, my poller was running with a centengine zombie process and the broker was stopped.

My retention files contain duplicate entries on comments.

To workaround the problem, I have to remove retention file but it should be great to fix this issue in the source code.

Please find below the complete error message :

[1444058089] [0x1be1840] error:   SQL: could not execute prepared statement: Cannot add or update a child row: a foreign key constraint fails (`centreon_storage`.`comments`, CONSTRAINT `comments_ibfk_1` FOREIGN KEY (`host_id`) REFERENCES `hosts` (`host_id`) ON DELETE CASCADE) QMYSQL3: Unable to execute statement

[1444058120] [0x1be1840] error:   SQL: could not execute prepared statement: Cannot add or update a child row: a foreign key constraint fails (`centreon_storage`.`comments`, CONSTRAINT `comments_ibfk_1` FOREIGN KEY (`host_id`) REFERENCES `hosts` (`host_id`) ON DELETE CASCADE) QMYSQL3: Unable to execute statement

Best regards.

**Comment by afouille** [comment]
Hello,

Your diagnostic is likely incorrect. This isn't a duplicate comment entry issue but an issue of the centengine zombie process having an outdated configuration and trying to inject incorrect data into Broker's stream. Thus broker tried to insert a comment that pertained to an invalid (because deleted) host, and it was correctly prevented to being inserted by the foreign key constraint.

There is nothing to do as everything work as intended, chiefly the constraint that prevented incorrect and outdated data from being inserted.

Best Regards,

**Comment by Staff457** [comment]
Dr.Agbalazzy, you are indeed a miracle worker. I have tried so many spell casters and psychic after my husband left me and told me to go and see the attorney for the divorce. And Dr.Agbalazzy was the only spell caster that could guarantee me an urgent 24hours spell casting.. and he assured me that my husband is coming back.. I trusted Dr. Agbalazzy, and its was more than a dream to see my husband back to my life with lots of apologies and love on the 24th of january 2015, And he was remorseful for all his wrong actions, For sure this is really a miracle temple. Olorunoduduwaspiritualtemple @g ma il.c om
